src/mlang/backend_compilers/dgfip_gen_files.ml

```diff
@@ -0,0 +1,1082 @@
+   GNU General Public License as published by the Free Software Foundation, either version 3 of the
+   License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+   even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+   General Public License for more details.
+
+   You should have received a copy of the GNU General Public License along with this program. If
+   not, see <https://www.gnu.org/licenses/>. *)
+
+module StringSet = Set.Make (String)
+module StringMap = Map.Make (String)
+
+type flags = {
+  (* -A *) nom_application : string;
+  (* iliad, pro, oceans, bareme, batch *)
+  (* -m *) annee_revenu : int;
+  (* -P *) flg_correctif : bool;
+  (* flg_correctif true by default, -P makes it false *)
+  (* -R *) flg_iliad : bool;
+  (* also implied by nom_application = "iliad"; disabled by -U *)
+  (* -R *) flg_pro : bool;
+  (* also implied by nom_application = "pro"; disabled by -U *)
+  (* -U *) flg_cfir : bool;
+  (* disabled by -R *)
+  (* -b *) flg_gcos : bool;
+  (* -b0 and -b1 ; disabled by -U and -R *)
+  (* -b *) flg_tri_ebcdic : bool;
+  (* -b1 only *)
+  (* -M *) flg_multithread : bool;
+  (* -s *) flg_short : bool;
+  (* -r *) flg_register : bool;
+  (* -O *) flg_optim_min_max : bool;
+  (* -X *) flg_extraction : bool;
+  (* -D *) flg_genere_libelle_restituee : bool;
+  (* -S *) flg_controle_separe : bool;
+  (* -I *) flg_controle_immediat : bool;
+  (* unused *)
+  (* -o *) flg_overlays : bool;
+  (* -Z *) flg_colors : bool;
+  (* -L *) flg_ticket : bool;
+  (* -t *) flg_trace : bool;
+  (* -g *) flg_debug : bool;
+  (* also implied by -t *)
+  (* -k *) nb_debug_c : int;
+  (* -x *)
+  xflg : bool;
+      (* Flags to deal with in a particular way : -c compilation mode -l link mode -v specify the
+         variable file (tgv.m) -e specify the error file (err.m) *)
+      (* Other flags, not used in makefiles -h dir_var_h -i flg_ident -C flg_compact -K
+         flg_optim_min_max -G flg_listing (+genere_cre = FALSE) -p flag_phase -f flg_ench_init -E
+         cvt_file -g flg_debug -a flg_api -T flg_trace_irdata *)
+}
+
+type gen_opt = {
+  with_verif : bool;
+  with_classe : bool;
+  with_priorite : bool;
+  with_categorie_TL : bool;
+  with_nat_code : bool;
+  with_cotsoc : bool;
+  with_ind_abat : bool;
+  with_acompte : bool;
+  with_avfisc : bool;
+  with_rapcat : bool;
+  with_sanction : bool;
+  with_modcat : bool;
+  with_liee : bool;
+  with_libelle : bool;
+  with_alias : bool;
+  with_type : bool;
+  with_type_donnee : bool;
+  with_primrest : bool;
+}
+
+type idx = {
+  out : int ref;
+  comp : int ref;
+  base : int ref;
+  inp : int ref;
+  ctxt : int ref;
+  fam : int ref;
+  rev : int ref;
+  revc : int ref;
+  var : int ref;
+  pen : int ref;
+}
+
+type var_subtype = Context | Family | Income | CorrIncome | Variation | Penality | Base | Computed
+
+type gen_type =
```

Comment:
You can maybe reuse and extend https://github.com/MLanguage/mlang/blob/master/src/mlang/m_frontend/mast.ml#L206 ?